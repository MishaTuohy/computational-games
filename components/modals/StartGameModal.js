import CreateGameButton from '@/components/buttons/CreateGameButton';
import JoinGameButton from '@/components/buttons/JoinGameButton';
import { useState } from 'react';
import Container from 'react-bootstrap/Container';
import Modal from 'react-bootstrap/Modal';

export default function StartGameModal({text, gameName}) {
    const [show, setShow] = useState(false);
    const handleClose = () => setShow(false);
    const handleShow = () => setShow(true);
  
    return (
        <>
            <div className="btn btn-warning" onClick={handleShow} data-testid="button">{text}</div>
            <Container data-testid="modal">
                <Modal show={show} onHide={handleClose} size="lg" aria-labelledby="contained-modal-title-vcenter" centered>
                    <Modal.Body className='login-modal'>
                        <Modal.Title>
                            {gameName}
                            <hr />
                        </Modal.Title>
                        <Container className='invite-container'>
                            <CreateGameButton game={gameName}/>
                            <JoinGameButton game={gameName}/>
                        </Container>
                    </Modal.Body>
                </Modal>
            </Container>
        </>
    );
}
