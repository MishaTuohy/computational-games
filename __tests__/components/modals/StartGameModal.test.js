import renderer from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react';
import StartGameModal from '@/components/modals/StartGameModal';

jest.mock('@/components/buttons/CreateGameButton', () => {
    return function MockCreateGameButton() {
        return <div data-testid="create-game-button">Create Game Button</div>;
    };
});

jest.mock('@/components/buttons/JoinGameButton', () => {
    return function MockJoinGameButton() {
        return <div data-testid="join-game-button">Join Game Button</div>;
    };
});

describe('StartGameModal', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<StartGameModal text="Start Game" gameName="Test Game" />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders the modal when the button is clicked', () => {
        const { getByTestId } = render(<StartGameModal text="Start Game" gameName="Test Game" />);
        const button = getByTestId('button');
        fireEvent.click(button);
        const tree = renderer.create(<StartGameModal text="Start Game" gameName="Test Game" />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('hides the modal when the close button is clicked', () => {
        const { getByTestId } = render(<StartGameModal text="Start Game" gameName="Test Game" />);
        const button = getByTestId('button');
        fireEvent.click(button);
        fireEvent.click(document);
        const tree = renderer.create(<StartGameModal text="Start Game" gameName="Test Game" />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
