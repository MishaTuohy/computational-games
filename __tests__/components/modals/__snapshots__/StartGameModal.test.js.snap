// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartGameModal hides the modal when the close button is clicked 1`] = `
[
  <div
    className="btn btn-warning"
    data-testid="button"
    onClick={[Function]}
  >
    Start Game
  </div>,
  <div
    className="container"
    data-testid="modal"
  />,
]
`;

exports[`StartGameModal renders correctly 1`] = `
[
  <div
    className="btn btn-warning"
    data-testid="button"
    onClick={[Function]}
  >
    Start Game
  </div>,
  <div
    className="container"
    data-testid="modal"
  />,
]
`;

exports[`StartGameModal renders the modal when the button is clicked 1`] = `
[
  <div
    className="btn btn-warning"
    data-testid="button"
    onClick={[Function]}
  >
    Start Game
  </div>,
  <div
    className="container"
    data-testid="modal"
  />,
]
`;
